packages/core/src/actions/execute.ts

```diff
@@ -750,28 +1006,241 @@ export const runEntireDeploymentProcess = async (
+}
+
+/**
+ * @notice Sorts an array of hex strings in ascending order. This function mutates the array.
+ */
+export const sortSigners = (arr: Array<TreeSigner>): void => {
+  arr.sort((a, b) => {
+    const aBigInt = BigInt(a.signer)
+    const bBigInt = BigInt(b.signer)
+
+    if (aBigInt < bBigInt) {
+      return -1
+    } else if (aBigInt > bBigInt) {
+      return 1
+    } else {
+      return 0
+    }
+  })
+}
+
+/**
+ * This is the primary function used for executing deployments in the deploy command, propose command simulation,
+ * and website backend. There is some logic in this function that may not be strictly necessary in all situations.
+ *
+ * We chose to reuse it in all situations because it is easier to maintain a single interface and because we want
+ * to improve the chance that bugs will get caught during the initial proposal simulation step rather than occuring
+ * when a deployment is actually happening on the website.
+ *
+ * The allow for this function to be reused in all three situations this function accepts a `deploymentContext` object
+ * which implements all the logic that needs to be different in each of the three situations.
+ *
+ * We aim to *only* use the `deploymentContext` for logic which absolutely cannot be shared. For example, logic for
+ * writing information about a deployment to the website backend is implemented in the `deploymentContext`.
+ *
+ * See documentation on the `DeploymentContext` type at the top of this file for more information of the specific fields
+ * and justification for why the logic implemented in them cannot be shared.
+ */
+export const compileAndExecuteDeployment = async (
+  deploymentContext: DeploymentContext
+): Promise<
+  | {
+      receipts: SphinxTransactionReceipt[]
+      batches: SphinxLeafWithProof[][]
+      finalStatus: BigInt
+      failureAction: HumanReadableAction | undefined
+    }
+  | undefined
+> => {
+  const { deployment } = deploymentContext
+  const networkName = deployment.networkName
+  const deploymentId = deployment.id
+  const rawCompilerConfigs = deployment.compilerConfigs
+  const { logger } = deploymentContext
+
+  if (!rawCompilerConfigs) {
+    deploymentContext.throwError(
+      `[Executor ${networkName}]: No compiler config stored for this deployment, should never happen`,
+      deployment.id,
+      networkName
+    )
+  }
+
+  logger?.info(`[Executor ${networkName}]: retrieving the deployment...`)
+  // Compile the bundle using either the provided localDeploymentId (when running the in-process
+  // executor), or using the Config URI
+  let merkleTree: SphinxMerkleTree
+  let compilerConfigs: Array<CompilerConfig>
+  let configArtifacts: ConfigArtifacts
+
+  // Handle if the config cannot be fetched
+  try {
+    ;({ merkleTree, compilerConfigs, configArtifacts } =
+      await buildDeploymentWithCompilerConfigs(rawCompilerConfigs!))
+  } catch (e: any) {
+    await deploymentContext.handleError(e, deployment, networkName, [])
+
+    return
```

Comment:
We only rethrow the error if `handleError` throws an error.  On the website, we just log an error and return. In long-lived processes, it's generally much better to just smoothly handle these sorts of things instead of actually throwing an error.